src/st/no1/e_mist_door.c

```diff
@@ -1,5 +1,78 @@
 // SPDX-License-Identifier: AGPL-3.0-or-later
-#include "common.h"
+#include "no1.h"
 
-// Mist door
-INCLUDE_ASM("st/no1/nonmatchings/e_mist_door", func_us_801D5A10);
+extern char D_psp_09265A88[];
+extern char D_psp_09265AA0[];
+extern char D_psp_09265AB8[];
+extern char D_psp_09265AD8[];
+extern char D_psp_09265AF0[];
+extern char D_psp_09265B08[];
+extern char D_psp_09265B20[];
+extern char D_psp_09265B38[];
+extern char D_psp_09265B58[];
+extern char D_psp_09265B70[];
+extern char D_psp_09265B88[];
+extern char D_psp_09265BA0[];
+extern char D_psp_09265BB8[];
+extern char D_psp_09265BD0[];
+extern char D_psp_09265BF0[];
+extern char D_psp_092753D8[];
+extern char* D_psp_0929A510;
+extern char* D_psp_0929A518;
+extern char* D_psp_0929A520;
+static char D_us_80183448[] = {
+    0x7C, 0x0E, 'M', 'i', 's', 't', ' ', 'c',  'o',  'u',
+    'l',  'd',  ' ', 'p', 'a', 's', 's', 0x81, 0x44, 0x00};
+static char D_us_8018345C[] = {
+    0x94, 0x1E, 'I', ' ', 'h',  'a',  'v',  'e',  ' ',  'n',  'o',
+    't',  'h',  'i', 'n', 'g',  ' ',  't',  'o',  ' ',  'd',  'o',
+    0x01, 'w',  'i', 't', 'h',  ' ',  't',  'h',  'i',  's',  ' ',
+    'w',  'a',  'l', 'l', 0x81, 0x44, 0x81, 0x44, 0x81, 0x44, 0x00};
+extern u16 g_EInitInteractable[];
+
+void EntityMistDoor(Entity* self) {
+    Entity* messageBox;
+
+    if (!self->step) {
+        InitializeEntity(g_EInitInteractable);
+        self->hitboxState = 1;
+        self->hitPoints = 0x7FFF;
+        self->attack = 0;
+        self->hitboxWidth = 0xC;
+        self->hitboxHeight = 0x18;
+    }
+    if (self->hitFlags) {
+        messageBox = AllocEntity(&g_Entities[224], &g_Entities[256]);
+        if (messageBox != NULL) {
+            CreateEntityFromCurrentEntity(E_SAVE_GAME_POPUP, messageBox);
+            messageBox->posX.i.hi = 0x80;
+            messageBox->posY.i.hi = 0xB0;
+#ifdef VERSION_PSP
+            D_psp_0929A520 = func_psp_092490A8(
+                D_psp_09265A88, D_psp_09265AD8, D_psp_09265AA0, D_psp_09265AF0,
+                D_psp_09265AB8);
+            D_psp_0929A518 = func_psp_092490A8(
+                D_psp_09265B88, D_psp_09265BD0, D_psp_09265BA0, D_psp_09265BF0,
+                D_psp_09265BB8);
+            D_psp_0929A510 = func_psp_092490A8(
+                D_psp_09265B08, D_psp_09265B58, D_psp_09265B20, D_psp_09265B70,
+                D_psp_09265B38);
+            if (g_PlayableCharacter == 2) {
```

Comment:
These have defines you can compare against (should probably be an enum, but I digress)

```
#define PLAYER_ALUCARD 0
#define PLAYER_RICHTER 1
#define PLAYER_MARIA 2
```

https://github.com/Xeeynamo/sotn-decomp/blob/master/include/game.h#L175-L177